### PR TITLE
List hints again

### PR DESCRIPTION
**Pull request by maciejka**
Choose which hints subset you are interested in:
```
cargo run --bin hints all | implemented | unimplemented
```

### CODE REVIEW

**Comment by LucasLvy** [comment]
you need to consider the base hints that are already implemented in the vm (you can use the starknet whitelisted hints file to get those)

**Comment by LucasLvy** [comment]
and i'd display the number of hints not the actual code

**Comment by maciejka** [comment]
> and i'd display the number of hints not the actual code

You can use `jq` to process it the way you want:
```
cargo run --bin hints unimplemented | jq length
```

**Comment by drspacemn** [comment]
Having trouble understanding the purpose of this? This is to display the meta-data for the current hint implementations in SNOS? 

Or simply to output their python strings in a large array?

Since this is meant to be used as a library I'm hesitant to add a top level `bin/` dir that doesn't add to the function of snos

**Review by drspacemn — COMMENTED** [review]
Having trouble understanding the purpose of this? This is to display the meta-data for the current hint implementations in SNOS?

Or simply to output their python strings in a large array?

Since this is meant to be used as a library I'm hesitant to add a top level bin/ dir that doesn't add to the function of snos

For hint accounting or metadata on dev progress I'd probably like to keep that to `scripts || tests || benches`

**Comment by maciejka** [comment]
It is a utility like a script. Ultimately it can be used in ci script to detect unimplemented or no longer needed hints.
The scripts dir seems most natural to me. Ok?

**Comment by drspacemn** [comment]
> The scripts dir seems most natural to me. Ok?

yes that sounds perfect to me

File: src/hints/mod.rs [hunk 1]
```diff
@@ -33,96 +36,53 @@ use crate::io::InternalTransaction;
...
-
-    let is_deprecated_hint = HintFunc(Box::new(is_deprecated));
-    hint_processor.add_hint(String::from(IS_DEPRECATED), Rc::new(is_deprecated_hint));
-
-    let os_context_segments_hint = HintFunc(Box::new(os_context_segments));
-    hint_processor.add_hint(String::from(OS_CONTEXT_SEGMENTS), Rc::new(os_context_segments_hint));
-
-    let selected_builtins_hint = HintFunc(Box::new(selected_builtins));
-    hint_processor.add_hint(String::from(SELECTED_BUILTINS), Rc::new(selected_builtins_hint));
+pub fn sn_hint_processor() -> BuiltinHintProcessor {
```
Comment (drspacemn) [inline src/hints/mod.rs:47]: #54

Will re-init how we set this handler up, so maybe we should hold off on this portion until that is merged

File: src/hints/mod.rs [hunk 2]
```diff
@@ -24,16 +24,65 @@ use starknet_api::deprecated_contract_class::ContractClass as DeprecatedContract
...
+    &mut ExecutionScopes,
+    &HashMap<String, HintReference>,
+    &ApTracking,
+) -> Result<HintExtension, HintError>;
+
+static EXTENSIVE_HINTS: [(&str, ExtensiveHintImpl); 1] = [(CHECK_DEPRECATED_CLASS_HASH, check_deprecated_class_hash)];
+
+pub struct SnosHintProcessor<'a> {
+    builtin_hint_proc: BuiltinHintProcessor,
+    hints: HashMap<&'a str, HintImpl>,
```
Comment (drspacemn) [inline src/hints/mod.rs:80]: Let's use `String` and get rid of the lifetime

**Review by drspacemn — APPROVED** [review]

**Review by drspacemn — CHANGES_REQUESTED** [review]
Let's use `String` to get rid of the lifetime

**Review by drspacemn — APPROVED** [review]

**Pull request merged**